pkg/cvo/availableupdates_test.go

```diff
@@ -73,61 +133,68 @@ func osusWithSingleConditionalEdge() (*httptest.Server, clusterconditions.Condit
     }
   ]
 }
-`, from, from, to, to, from, to, to)
-	}))
+`, data.from.version, data.from.image, data.to.version, data.to.image, data.from.version,
+			data.to.version, data.to.version)
+	})), &params
+}
 
-	updates := []configv1.ConditionalUpdate{
-		{
-			Release: configv1.Release{Version: to, Image: "payload/" + to},
-			Risks: []configv1.ConditionalUpdateRisk{
-				{
-					URL:     "https://example.com/" + to,
-					Name:    "FourFiveSix",
-					Message: "Four Five Five is just fine",
-					MatchingRules: []configv1.ClusterCondition{
-						{
-							Type:   "PromQL",
-							PromQL: &configv1.PromQLClusterCondition{PromQL: "this is a query"},
-						},
-					},
-				},
-			},
-			Conditions: []metav1.Condition{
-				{
-					Type:               "Recommended",
-					Status:             metav1.ConditionFalse,
-					Reason:             "FourFiveSix",
-					Message:            "Four Five Five is just fine https://example.com/" + to,
-					LastTransitionTime: metav1.Now(),
-				},
-			},
+type testFixture struct {
+	// Mock osus server
+	osus *httptest.Server
+	// Query parameters used in the last call to the osus server
+	lastQueryParams *url.Values
+
+	// Expected updates after evaluation of the mockCondition
+	expectedConditionalUpdates []configv1.ConditionalUpdate
+	mockCondition              clusterconditions.Condition
+
+	currentRelease release
+}
+
+// osusWithSingleConditionalEdge helper returns:
+//  1. mock osus server that serves a simple conditional path between two versions.
+//  2. mock condition that always evaluates to match
+//  3. expected []ConditionalUpdate data after evaluation of the data served by mock osus server
+//     (assuming the mock condition (2) was used)
+//  4. current version of the cluster that would issue the request to the mock osus server
+//  5. current image of the cluster
+//  6. query parameters used in the last call to the osus server
+func osusWithSingleConditionalEdge(data conditionalEdgeTestData) testFixture {
+	server, lastQueryParams := newMockOSUSServer(data)
+	expectedUpdates, mockPromql := expectedConditionalUpdatesFor(data)
+	return testFixture{
+		osus:                       server,
+		lastQueryParams:            lastQueryParams,
+		expectedConditionalUpdates: expectedUpdates,
+		mockCondition:              mockPromql,
+		currentRelease: release{
+			version: data.from.version,
+			image:   data.from.image,
 		},
 	}
-	mockPromql := &mock.Mock{
-		ValidQueue: []error{nil},
-		MatchQueue: []mock.MatchResult{{Match: true, Error: nil}},
-	}
-
-	return osus, mockPromql, updates, from
 }
 
-func newOperator(url, version string, promqlMock clusterconditions.Condition) (*availableUpdates, *Operator) {
-	currentRelease := configv1.Release{Version: version, Image: "payload/" + version}
+func newOperator(url string, cluster release, promqlMock clusterconditions.Condition, arch string) (*availableUpdates, *Operator) {
```

Comment:
nit: Target CPU architecture is a property of the release payload, so it could be an `architecture` property on the `release` structure.  That way it could get passed in via the `cluster` argument, instead of needing a 4th argument here.  But 🤷, this is just minor shuffling, and no worries from me if you leave this as it stands.

Reply:
Thanks, a good observation. This would require some further refactoring. I am not sure how to do it nicely and quickly. I will try to create a follow-up PR with this change (see https://github.com/openshift/cluster-version-operator/pull/1255#discussion_r2566502465).